integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+      { user: userName, deviceName },
+      true
+  );
+  const params = {
+	    deviceId: getCreds(deviceName).id
+  };
+  const fullUrl = v1ApiPath('devices/users', params);
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "GET",
+        url: fullUrl,
+        body: null
+      },
+      userName,
+      statusCode
+    ).then((response) => {
+       if(statusCode==null || statusCode==200) {
+	  // API returns devices: [ groupname: ..., devicename: ..., saltId, ..., Group.groupName: ... ]
+	  // sort users and expected users to ensure order is the same
+          var users=sortArrayOn(response.body.rows, 'username');
+          expectedUsers=sortArrayOn(expectedUsers, 'username');
+	  expect(users.length).to.equal(expectedUsers.length);
+
+          for (var index=0; index < expectedUsers.length; index++) {
+		  expect(users[index].id).to.equal(expectedUsers[index].id);
+		  expect(users[index].username).to.equal(expectedUsers[index].username);
+		  expect(users[index].relation).to.equal(expectedUsers[index].relation);
+		  expect(users[index].admin).to.equal(expectedUsers[index].admin);
+		  expect(users[index].email).to.equal(expectedUsers[index].email);
+	  }
+       };
+  });
+});
+
+Cypress.Commands.add(
+  "apiAddUserToDevice",
+  (deviceAdminUser: string, userName: string, device: string, admin: boolean = false, statusCode: number = 200) => {
+    logTestDescription(
+      `${deviceAdminUser} Adding user '${userName}' to device '${device}'`,
+      { user: userName, device }
+    );
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "POST",
+        url: v1ApiPath("devices/users"),
+        body: {
+          deviceId: getCreds(device).id,
+          admin: admin,
+          username: getTestName(userName)
+        }
+      },
+      deviceAdminUser,
+      statusCode
+    );
+  }
+);
+
+Cypress.Commands.add(
+  "apiRemoveUserFromDevice",
+  (deviceAdminUser: string, userName: string, device: string, statusCode: number = 200) => {
+    logTestDescription(
+      `${deviceAdminUser} Removing user '${userName}' to device '${device}'`,
+      { user: userName, device }
+    );
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "DELETE",
+        url: v1ApiPath("devices/users"),
+        body: {
+          deviceId: getCreds(device).id,
+          admin: "false",
+          username: getTestName(userName)
+        }
+      },
+      deviceAdminUser,
+      statusCode
+    );
+  }
+);
+
+
+
+
+Cypress.Commands.add("apiUploadRecording", (cameraName, id) => {
+  const recordingsUrl = v1ApiPath('recordings');
+  const deviceName = getTestName(cameraName);
+
+  const data = {
+       type: 'thermalRaw'
+  };
+
+  const fileName="bird_"+id+".cptv";
+  const uniqueName=deviceName+"_"+id;
+  const aliasName=deviceName+"_a"+id;
+
+
+  const creds = getCreds(cameraName);
+  uploadFileRequest(fileName, uniqueName, aliasName, recordingsUrl, data, creds)
```

Comment:
does this work? i would of thought aliasName shoul dbe the deviceName? and it doesn't look like uniqueName is even used. should be deleted?

Reply:
Noted.  Part of browser testing code.  Will revisit this when browser testing code is updated. This function and the one used by the API tests in fileUpload.ts do the same job and need to be combined.

Reply:
Deleted.  Duplicate function left over from browser test code. Reworked browser tests to use the other version of this function.